src/nchan_module.c

```diff
@@ -296,6 +298,68 @@ ngx_int_t nchan_stub_status_handler(ngx_http_request_t *r) {
   return ngx_http_output_filter(r, &out);
 }
 
+static ngx_int_t redis_stats_callback(ngx_int_t rc, void *d, void *pd) {
+  redis_nodeset_command_stats_t *stats = d;
+  ngx_http_request_t            *r = pd;
+  ngx_str_t                      content_type_plain = ngx_string("text/plain");
+  ngx_str_t                      content_type_json = ngx_string("application/json");
+  
+  
+  if(stats->error) {
+    nchan_respond_cstring(r, NGX_HTTP_INTERNAL_SERVER_ERROR, &content_type_plain, stats->error, 1);
+    return NGX_OK;
+  }
+  
+  if(stats->count == 0 || stats->stats == NULL) {
+    nchan_respond_cstring(r, NGX_HTTP_INTERNAL_SERVER_ERROR, &content_type_plain, "weird error getting status data", 1);
+    return NGX_OK;
+  }
+  
+  ngx_chain_t *body = redis_nodeset_stats_response_body_chain_palloc(stats, r->pool);
+  if(!body) {
+    nchan_respond_cstring(r, NGX_HTTP_INTERNAL_SERVER_ERROR, &content_type_plain, "failed to allocate response body", 1);
+    return NGX_OK;
+  }
+  
+  r->headers_out.content_type = content_type_json;
+  nchan_respond_status(r, NGX_HTTP_OK, NULL, body, 1);
```

Comment:
Yes, but the failure is handled and logged by Nginx internals. No additional action needs to be taken on failure -- indeed, apart from logging the error, nothing can be done to deliver the failure information to the HTTP client -- because the response failed.